apps/admin-gui/src/app/shared/components/security-teams-list/security-teams-list.component.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  AfterViewInit,
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  Output,
+  SimpleChanges,
+  ViewChild
+} from '@angular/core';
+import { SecurityTeam } from '@perun-web-apps/perun/openapi';
+import { MatSort } from '@angular/material/sort';
+import { SelectionModel } from '@angular/cdk/collections';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { TABLE_ITEMS_COUNT_OPTIONS } from '@perun-web-apps/perun/utils';
+
+@Component({
+  selector: 'app-security-teams-list',
+  templateUrl: './security-teams-list.component.html',
+  styleUrls: ['./security-teams-list.component.scss']
+})
+export class SecurityTeamsListComponent implements AfterViewInit, OnChanges {
+
+  constructor() { }
+
+  @ViewChild(MatSort, { static: true }) set matSort(ms: MatSort) {
+    this.sort = ms;
+    this.setDataSource();
+  }
+
+  @Input()
+  securityTeams: SecurityTeam[] = [];
+  @Input()
+  selection = new SelectionModel<SecurityTeam>(true, []);
+  @Input()
+  filterValue: string;
+  @Input()
+  pageSize = 10;
+
+  @Output()
+  page: EventEmitter<PageEvent> = new EventEmitter<PageEvent>();
+
+  private sort: MatSort;
+
+  displayedColumns: string[] = ['select', 'id', "name", "description"];
+  dataSource: MatTableDataSource<SecurityTeam>;
+
+  exporting = false;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  pageSizeOptions = TABLE_ITEMS_COUNT_OPTIONS;
+
+  ngOnChanges(changes: SimpleChanges) {
+    this.dataSource = new MatTableDataSource<SecurityTeam>(this.securityTeams);
+    this.setDataSource();
+    this.dataSource.filter = this.filterValue;
+  }
+
+
+  setDataSource() {
+    if (!!this.dataSource) {
+      this.dataSource.sort = this.sort;
+      this.dataSource.sortingDataAccessor = (item, property) => {
+        switch (property) {
+          case 'name' :{
+            if (item.name) {
+              return item.name;
+            }
+            break;
+          }
+          default: return item[property];
+        }
+      };
```

Comment:
This doesnt have to be here always. Only when the sorting is not working. And in this component we cant decide that so remove it please